tools/config/manage-dependencies.ts

```diff
@@ -0,0 +1,1163 @@
+          },
+        };
+      }
+      imp.files.forEach((file) => {
+        lib.externalDependencies[dependency].files.add(file);
+      });
+      if (imp.usageIn.lib) {
+        lib.externalDependencies[dependency].usageIn.lib = true;
+      }
+      if (imp.usageIn.schematics) {
+        lib.externalDependencies[dependency].usageIn.schematics = true;
+      }
+      if (imp.usageIn.schematicsSpec) {
+        lib.externalDependencies[dependency].usageIn.schematicsSpec = true;
+      }
+      if (imp.usageIn.spec) {
+        lib.externalDependencies[dependency].usageIn.spec = true;
+      }
+    });
+    Object.values(lib.scssImports).forEach((imp) => {
+      let dependency;
+      let dep;
+      if (imp.importPath.startsWith('~')) {
+        dep = imp.importPath.substring(1);
+      } else {
+        dep = imp.importPath.substring('node_modules/'.length);
+      }
+      if (dep.startsWith('@')) {
+        const [scope, name] = dep.split('/');
+        dependency = `${scope}/${name}`;
+      } else {
+        const [name] = dep.split('/');
+        dependency = `${name}`;
+      }
+
+      if (!lib.externalDependencies[dependency]) {
+        lib.externalDependencies[dependency] = {
+          dependency,
+          files: new Set<string>(),
+          usageIn: {
+            lib: false,
+            spec: false,
+            schematics: false,
+            schematicsSpec: false,
+            styles: false,
+          },
+        };
+      }
+      imp.files.forEach((file) => {
+        lib.externalDependencies[dependency].files.add(file);
+      });
+      lib.externalDependencies[dependency].usageIn.styles = true;
+    });
+  });
+}
+
+/**
+ * Get external dependencies from imports
+ */
+function checkIfWeHaveAllDependenciesInPackageJson(
+  libraries: Record<string, LibraryWithDependencies>,
+  packageJson: PackageJson,
+  options: ProgramOptions
+): void {
+  if (!options.fix) {
+    const allDeps = {
+      ...packageJson.devDependencies,
+      ...packageJson.dependencies,
+    };
+    const errors = [];
+    reportProgress(`Checking for missing dependencies in root ${PACKAGE_JSON}`);
+    Object.values(libraries).forEach((lib) => {
+      Object.values(lib.externalDependencies).forEach((dep) => {
+        if (
+          !dep.dependency.startsWith(`${SPARTACUS_SCOPE}/`) &&
+          !Object.keys(allDeps).includes(dep.dependency)
+        ) {
+          errors.push(
+            `Missing \`${chalk.bold(
+              dep.dependency
+            )}\` dependency that is used directly in \`${chalk.bold(
+              lib.name
+            )}\`.`
+          );
+        }
+      });
+    });
+    if (errors.length) {
+      error(PACKAGE_JSON, errors, [
+        `All dependencies that are directly referenced should be specified as \`${chalk.bold(
+          'dependencies'
+        )}\` or \`${chalk.bold('devDependencies')}\`.`,
+        `Install them with \`${chalk.bold(
+          'yarn add <dependency-name> (--dev)'
```

Comment:
As `--dev` is an optional argument, let's use the common syntax for optional args: `[--dev]`